### PR TITLE
[IMP] pos: Negative payment (refund) on touchscreen

### PR DESCRIPTION
**Pull request by mgh-odoo**
currently, if a user wants to return fund to the customer.
we don't have a button to bill negatively. that will fixed.

Task-ID: 1929650
Closes: #31054

Task:https://www.odoo.com/web#id=1929650&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720
Pad:https://pad.odoo.com/p/r.5745606edc51c9662beff81d277af3e3

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/xml/pos.xml [hunk 1]
```diff
@@ -1553,7 +1553,7 @@
                     </tr>
                 </t>
                 <tr>
-                    <t t-if="order.get_total_discount() > 0">
+                    <t t-if="order.get_total_discount()">
```
Comment (switch87) [inline addons/point_of_sale/static/src/xml/pos.xml]: I agree this look better, but this is totally out of scope for this task, could you remove this out of the commit?

File: addons/point_of_sale/static/src/js/models.js [hunk 2]
```diff
@@ -2501,8 +2501,9 @@ exports.Order = Backbone.Model.extend({
     },
     get_total_discount: function() {
         return round_pr(this.orderlines.reduce((function(sum, orderLine) {
-            sum += (orderLine.get_unit_price() * (orderLine.get_discount()/100) * orderLine.get_quantity());
-            sum += ((orderLine.get_lst_price() - orderLine.get_unit_price()) * orderLine.get_quantity());
+            var unit_price = orderLine.get_unit_price();
```
Comment (switch87) [inline addons/point_of_sale/static/src/js/models.js]: get_quantity and Math.abs(unit_price) could also be set in a separate variable.

File: addons/point_of_sale/static/src/xml/pos.xml (hunk 1 above)
Comment (mgh-odoo) [reply addons/point_of_sale/static/src/xml/pos.xml]: hello, @switch87 
if we remove this then not need to use Math.abs() and Math.sign() in models.js https://github.com/odoo/odoo/pull/31054/commits/a51cf9496b7e591c627811a8ef5f09e53019b955#diff-1c14accbe350f7a177b3e81e5846861bR2506
I have added to show returned product price with a previously applied discount minus.
only thing is left now is, replace "Clear" to "+/-" is it ok?
Thanks

Comment (switch87) [reply addons/point_of_sale/static/src/xml/pos.xml]: @mgh-odoo can you record an example pre- and post-commit on runbot of this? The task was indeed as simple as replacing this key apparently. But if this extra work is indeed an issue that should be fixed, you can split the commit into one  [IMP] and one [FIX] and your work would not be for nothing :)

Comment (mgh-odoo) [reply addons/point_of_sale/static/src/xml/pos.xml]: hello, @switch87 
I changed commit based on your suggestion and 
also recorded a video about [pre-commit](https://drive.google.com/open?id=1dKH5g7RZARmsN_0p43FaGNw7mGDNvyqf) and [post-commit](https://drive.google.com/open?id=17xgvw3Ov0HqViKy7iaLnyGKG0lMJ44_x) 
if it's out of scope, let me know I'll remove that one :)

Comment (switch87) [reply addons/point_of_sale/static/src/xml/pos.xml]: @mgh-odoo yes it's out of scope for this task. I think it is not a bad idea, so make it a new pr for it and ask @lap-odoo for her opinion.

Comment (mgh-odoo) [reply addons/point_of_sale/static/src/xml/pos.xml]: @switch87  done. :+1:

**Comment by switch87** [comment]
@qle-odoo r+ please

**Comment by qle-odoo** [comment]
@robodoo r+
